Makefile

```diff
@@ -96,20 +97,23 @@ verify.tidy:
 	./hack/verify-tidy.sh
 
 .PHONY: verify.manifests
-verify.manifests:
+verify.manifests: kustomize
 	./hack/verify-manifests.sh
 
 # ------------------------------------------------------------------------------
 # Build - Manifests
 # ------------------------------------------------------------------------------
 
 .PHONY: manifests
-manifests: controller-gen ## Generate WebhookConfiguration, ClusterRole and CustomResourceDefinition objects.
-	$(CONTROLLER_GEN) $(CRD_OPTIONS) rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
-	go run hack/generators/manifests/main.go --directory config/crd/bases/
+manifests: manifests.crds manifests.single
```

Comment:
`manifests.single-v2` is not an existing `Makefile` target, so making this change verbatim would break the makefile:

```shell
make: *** No rule to make target 'manifests.single-v2', needed by 'manifests'.  Stop.
```

Do you mean to say you would like to see the target name renamed to `manifests.single-v2` entirely?

I would hesitate to make this change since we don't have any other manifest generator right now: we are keeping the older (v1) manifests around manually for historical purposes.

Reply:
both manifests.crds and manifests.single-v2 are pure symbols defined within Makefile, the request add on clarity into the code.

Reply:
`manifests.single-v2` is _not_ already a defined symbol. Do you want it to be? I'm opposed to adding the `-v2` specificity because there's no other implementation.

Reply:
Per https://github.com/Kong/kubernetes-ingress-controller/issues/1685#issuecomment-900519403 2.0 GA will place the 2.0 single manifests under `deploy/single`. The separate single-v2 stuff we have is temporary during the beta; once we're out of that we'll have no need for separate 1.x and 2.x manifests (if you still need 1.x you'll go to a 1.x tag rather than main).

As such I don't think we'd want to change the Makefile target name, since we'll ultimately just flip it back anyway.

Reply:
since atm, the code (hack/deploy/build-single-manifests.sh) still work on single-v2 directory, to be clear, use single-v2 until we remove single-v2. 
otherwise, confused.

Reply:
I'm confused about how this thread went: it was marked resolved and commits were pushed to this branch to make the change that was still actively in question without the conversation appearing to be at all resolved.

For now I've dropped those commits in a rebase and I'm marking this unresolved again so we can continue that conversation.

Reply:
After 2.0, we should delete both single and single-v2, and move all yaml into manifests directory. 
Currently, change the target name to single-v2 gives us correct information and data - script  ./hack/deploy/build-single-manifests.sh is publishing data to manifests/single-v2, not manifests/single, you wont find any update within manifests/single

Reply:
> you wont find any update within manifests/single

Correct, but this is intentional currently.

Reply:
Intentionally, better fix this in this PR.

Reply:
Whether the current (autogenerated "single-v2" + deep copy into "single") directory structure under `deploy/` is right or wrong is a discussion worth having, but probably not on this PR because the scope appears very different.

I suggest the following resolution if everybody's on board:
- on this PR keep the name of the target as is
- if necessary, create an issue (unrelated to this PR, but possibly linking to this discussion) explaining the proposed change of naming.

Sounds good? @shaneutt @ccfishk